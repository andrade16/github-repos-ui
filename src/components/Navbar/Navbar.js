import React from "react";
import {GoLogoGithub} from 'react-icons/go';
import "./Navbar.scss";

const Navbar = () => {

    return (
        <header id="header" className="navbar">
            <a className="logo" href="https://github.com/">
                <GoLogoGithub style={{width: 50, height: 50, color: 'white'}}/>
            </a>

            <nav className="navigation">{"Nav content"}</nav>
        </header>
    );
};

export default Navbar;
