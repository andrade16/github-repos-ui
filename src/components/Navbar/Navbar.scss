@import "src/utils/variables";

.navbar {
  display: $flex-display;
  justify-content: space-between;
  align-items: $content-center;
  padding: 0 10px 0 10px;
  width: 100%;
  min-height: 10vh;
  box-sizing: border-box;
  background-color: #24292e;
  transition: all 0.7s ease-in;
  position: fixed;
  top: 0;
  left: 0;
  z-index: 1;
  box-shadow: 0 0 15px 0 rgba(0, 0, 0, 0.51);

  .app-title {
    color: white;
    margin-left: 28px;
    margin-right: 28px;
  }

  .logo {
    cursor: $pointer-cursor;

    #github-icon {
      color: $white;
      width: 60px;
      height: 50px;
    }
  }

  .navigation {
    width: 100px;
    display: flex;
    justify-content: space-evenly;

    @media (max-width: 340px) {
      flex-direction: $flex-column;
    }

    a {
      cursor: $pointer-cursor;
      #email-icon,
      #linkedin-icon {
        width: 30px;
        height: 30px;
        color: $white;
      }
    }
  }
}
