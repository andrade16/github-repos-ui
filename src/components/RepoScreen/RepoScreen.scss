@import "src/utils/variables";

.container {
  height: 90vh;
  margin: 4rem;
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
  grid-template-rows: repeat(3, 1fr);
  grid-gap: 40px;

  @media (max-width: 400px) {
    margin-left: 0;
    margin-right: 0;
  }
}

.spacer {
  width: 100%;
  height: 10vh;
}

.scroll-top-container {
  display: $flex-display;
  justify-content: flex-end;
}
