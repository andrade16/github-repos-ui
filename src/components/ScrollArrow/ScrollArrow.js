import React, { useState } from "react";
import { Y_OFFSET_THRESHOLD } from "../../utils/constants";
import { FaArrowCircleUp } from "react-icons/fa";
import "./ScrollArrow.scss";

const ScrollArrow = () => {
  const [showScroll, setShowScroll] = useState(false);

  const checkScrollTop = () => {
    if (!showScroll && window.pageYOffset > Y_OFFSET_THRESHOLD) {
      setShowScroll(true);
    } else if (showScroll && window.pageYOffset <= Y_OFFSET_THRESHOLD) {
      setShowScroll(false);
    }
  };

  const scrollTop = () => {
    window.scrollTo({ top: 0, behavior: "smooth" });
  };

  window.addEventListener("scroll", checkScrollTop);

  return (
    <FaArrowCircleUp
      className="scrollTop"
      onClick={scrollTop}
      style={{ height: 40, display: showScroll ? "flex" : "none" }}
    />
  );
};

export default ScrollArrow;
