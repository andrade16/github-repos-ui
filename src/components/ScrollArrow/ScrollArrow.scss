@import "src/utils/variables";

.scrollTop {
  position: fixed;
  width: 64px;
  bottom: 20px;
  align-items: $content-center;
  height: 20px;
  justify-content: $content-center;
  z-index: 1000;
  cursor: $pointer-cursor;
  animation: fadeIn 0.3s;
  transition: opacity 0.4s;
  opacity: 0.5;
}

.scrollTop:hover {
  opacity: 1;
}

@keyframes fadeIn {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 0.5;
  }
}
