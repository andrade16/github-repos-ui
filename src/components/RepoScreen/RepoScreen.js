import React, { useState, useEffect } from "react";
import { FaGithub } from "react-icons/fa";
import { FiStar } from "react-icons/fi";
import { getStarredRepos, getCommits } from "../../services/githubService";
import { formatDateByDay } from "../../utils/helpers";
import Navbar from "../Navbar/Navbar";
import Loader from "../Loader/Loader";
import Card from "../Card/Card";
import Modal from "../Modal/Modal";
import ScrollArrow from "../ScrollArrow/ScrollArrow";
import List from "../List/List";
import EmptyDisplay from "../EmptyDisplay/EmptyDisplay";
import moment from "moment";
import "./RepoScreen.scss";

const RepoScreen = () => {
  const [repos, setRepos] = useState([]);
  const [commits, setCommits] = useState([]);
  const [show, setShow] = useState(false);
  const [isLoading, setIsLoading] = useState(false);
  const today = moment().toISOString(true);

  useEffect(() => {
    setIsLoading(true);
    getStarredRepos().then((data) => {
      setRepos(data.items);
      setIsLoading(false);
    });
  }, []);

  const handleCardClick = (dataObj) => {
    setShow(true);
    setIsLoading(true);
    getCommits(dataObj).then((data) => {
      setCommits(data);
      setIsLoading(false);
    });
  };

  const onModalClose = () => {
    setShow(false);
    setCommits([]);
  };

  let content = <Loader message="Loading..." />;
  if (repos.length) {
    content = repos.map((repo) => (
      <Card
        key={repo.id}
        title={repo.full_name}
        link={repo.html_url}
        subtitle={`Stars: ${repo.stargazers_count}`}
        avatarUrl={repo.owner.avatar_url}
        icon={<FaGithub size={150} style={{ marginTop: 5 }} />}
        subIcon={<FiStar size={15} fill="yellow" />}
        dataObj={repo}
        onClick={handleCardClick}
      />
    ));
  }

  let modalContent;
  if (isLoading) {
    modalContent = <Loader message="Getting commits..." />;
  } else if (commits.length > 0) {
    modalContent = <List data={commits} />;
  } else {
    modalContent = <EmptyDisplay message="No recent commits to display" />;
  }

  return (
    <>
      <Navbar />
      <div className="spacer">&nbsp;</div>

      <div className="container">
        {content}

        <Modal
          show={show}
          onClose={onModalClose}
          header={`Commits on ${formatDateByDay(today)}`}
        >
          {modalContent}
        </Modal>
      </div>

      <div className="scroll-top-container">
        <ScrollArrow />
      </div>
    </>
  );
};

export default RepoScreen;
