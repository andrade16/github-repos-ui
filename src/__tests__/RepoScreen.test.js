import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import RepoScreen from "../components/RepoScreen/RepoScreen";
import Navbar from "../components/Navbar/Navbar";
import ScrollArrow from "../components/ScrollArrow/ScrollArrow";

Enzyme.configure({ adapter: new Adapter() });

describe("<RepoScreen/>", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<RepoScreen />);
  });

  it("renders the <Navbar /> component", () => {
    expect(wrapper.find(Navbar)).toHaveLength(1);
  });

  it("renders the spacer div", () => {
    expect(wrapper.find(".spacer")).toHaveLength(1);
  });

  it("renders the ScrollArrow component", () => {
    expect(wrapper.find(ScrollArrow)).toHaveLength(1);
  });
});
