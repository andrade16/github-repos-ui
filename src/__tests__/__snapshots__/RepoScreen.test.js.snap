// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepoScreen/> Renders <RepoScreen/> correctly 1`] = `
Array [
  <header
    className="navbar"
  >
    <a
      className="logo"
      href="https://github.com/"
    >
      <svg
        fill="currentColor"
        height="1em"
        id="github-icon"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 45 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18.53 12.03h-.02c.009 0 .015.01.024.011h.006l-.01-.01zm.004.011c-.093.001-.327.05-.574.05-.78 0-1.05-.36-1.05-.83V8.13h1.59c.09 0 .16-.08.16-.19v-1.7c0-.09-.08-.17-.16-.17h-1.59V3.96c0-.08-.05-.13-.14-.13h-2.16c-.09 0-.14.05-.14.13v2.17s-1.09.27-1.16.28c-.08.02-.13.09-.13.17v1.36c0 .11.08.19.17.19h1.11v3.28c0 2.44 1.7 2.69 2.86 2.69.53 0 1.17-.17 1.27-.22.06-.02.09-.09.09-.16v-1.5a.177.177 0 0 0-.146-.18zM42.23 9.84c0-1.81-.73-2.05-1.5-1.97-.6.04-1.08.34-1.08.34v3.52s.49.34 1.22.36c1.03.03 1.36-.34 1.36-2.25zm2.43-.16c0 3.43-1.11 4.41-3.05 4.41-1.64 0-2.52-.83-2.52-.83s-.04.46-.09.52c-.03.06-.08.08-.14.08h-1.48c-.1 0-.19-.08-.19-.17l.02-11.11c0-.09.08-.17.17-.17h2.13c.09 0 .17.08.17.17v3.77s.82-.53 2.02-.53l-.01-.02c1.2 0 2.97.45 2.97 3.88zm-8.72-3.61h-2.1c-.11 0-.17.08-.17.19v5.44s-.55.39-1.3.39-.97-.34-.97-1.09V6.25c0-.09-.08-.17-.17-.17h-2.14c-.09 0-.17.08-.17.17v5.11c0 2.2 1.23 2.75 2.92 2.75 1.39 0 2.52-.77 2.52-.77s.05.39.08.45c.02.05.09.09.16.09h1.34c.11 0 .17-.08.17-.17l.02-7.47c0-.09-.08-.17-.19-.17zm-23.7-.01h-2.13c-.09 0-.17.09-.17.2v7.34c0 .2.13.27.3.27h1.92c.2 0 .25-.09.25-.27V6.23c0-.09-.08-.17-.17-.17zm-1.05-3.38c-.77 0-1.38.61-1.38 1.38 0 .77.61 1.38 1.38 1.38.75 0 1.36-.61 1.36-1.38 0-.77-.61-1.38-1.36-1.38zm16.49-.25h-2.11c-.09 0-.17.08-.17.17v4.09h-3.31V2.6c0-.09-.08-.17-.17-.17h-2.13c-.09 0-.17.08-.17.17v11.11c0 .09.09.17.17.17h2.13c.09 0 .17-.08.17-.17V8.96h3.31l-.02 4.75c0 .09.08.17.17.17h2.13c.09 0 .17-.08.17-.17V2.6c0-.09-.08-.17-.17-.17zM8.81 7.35v5.74c0 .04-.01.11-.06.13 0 0-1.25.89-3.31.89-2.49 0-5.44-.78-5.44-5.92S2.58 1.99 5.1 2c2.18 0 3.06.49 3.2.58.04.05.06.09.06.14L7.94 4.5c0 .09-.09.2-.2.17-.36-.11-.9-.33-2.17-.33-1.47 0-3.05.42-3.05 3.73s1.5 3.7 2.58 3.7c.92 0 1.25-.11 1.25-.11v-2.3H4.88c-.11 0-.19-.08-.19-.17V7.35c0-.09.08-.17.19-.17h3.74c.11 0 .19.08.19.17z"
          fillRule="evenodd"
        />
      </svg>
    </a>
    <h2
      className="app-title"
    >
      Top 100 Repositories
    </h2>
    <nav
      className="navigation"
    >
      <a
        href="mailto:andrade.william61@gmail.com"
      >
        <svg
          fill="currentColor"
          height="1em"
          id="email-icon"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 110.8V792H136V270.8l-27.6-21.5 39.3-50.5 42.8 33.3h643.1l42.8-33.3 39.3 50.5-27.7 21.5zM833.6 232L512 482 190.4 232l-42.8-33.3-39.3 50.5 27.6 21.5 341.6 265.6a55.99 55.99 0 0 0 68.7 0L888 270.8l27.6-21.5-39.3-50.5-42.7 33.2z"
          />
        </svg>
      </a>
      <a
        href="https://www.linkedin.com/in/william-andrade/"
      >
        <svg
          fill="currentColor"
          height="1em"
          id="linkedin-icon"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M847.7 112H176.3c-35.5 0-64.3 28.8-64.3 64.3v671.4c0 35.5 28.8 64.3 64.3 64.3h671.4c35.5 0 64.3-28.8 64.3-64.3V176.3c0-35.5-28.8-64.3-64.3-64.3zm0 736c-447.8-.1-671.7-.2-671.7-.3.1-447.8.2-671.7.3-671.7 447.8.1 671.7.2 671.7.3-.1 447.8-.2 671.7-.3 671.7zM230.6 411.9h118.7v381.8H230.6zm59.4-52.2c37.9 0 68.8-30.8 68.8-68.8a68.8 68.8 0 1 0-137.6 0c-.1 38 30.7 68.8 68.8 68.8zm252.3 245.1c0-49.8 9.5-98 71.2-98 60.8 0 61.7 56.9 61.7 101.2v185.7h118.6V584.3c0-102.8-22.2-181.9-142.3-181.9-57.7 0-96.4 31.7-112.3 61.7h-1.6v-52.2H423.7v381.8h118.6V604.8z"
          />
        </svg>
      </a>
    </nav>
  </header>,
  <div
    className="spacer"
  >
    Â 
  </div>,
  <div
    className="container"
  >
    <div
      className="loader-container"
    >
      <div
        className="loader"
      />
      <span
        className="loading-text"
      >
        Loading...
      </span>
    </div>
    <div
      className="modal "
      onClick={[Function]}
    >
      <div
        className="modal-content"
        onClick={[Function]}
      >
        <div
          className="modal-header"
        >
          <h4
            className="title"
            style={
              Object {
                "margin": 0,
              }
            }
          >
            Commits on Feb 23, 2021
          </h4>
        </div>
        <div
          className="modal-body"
        >
          <div
            className="empty-display-container"
          >
            <svg
              fill="currentColor"
              height="1em"
              id="warning-icon"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              version="1.1"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 1.45l6.705 13.363h-13.409l6.705-13.363zM8 0c-0.345 0-0.69 0.233-0.951 0.698l-6.829 13.611c-0.523 0.93-0.078 1.691 0.989 1.691h13.583c1.067 0 1.512-0.761 0.989-1.691h0l-6.829-13.611c-0.262-0.465-0.606-0.698-0.951-0.698v0z"
              />
              <path
                d="M9 13c0 0.552-0.448 1-1 1s-1-0.448-1-1c0-0.552 0.448-1 1-1s1 0.448 1 1z"
              />
              <path
                d="M8 11c-0.552 0-1-0.448-1-1v-3c0-0.552 0.448-1 1-1s1 0.448 1 1v3c0 0.552-0.448 1-1 1z"
              />
            </svg>
            <div
              className="empty-display-message"
            >
              No recent commits to display
            </div>
          </div>
        </div>
        <div
          className="modal-footer"
        >
          <button
            className="button"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#dc3545",
                "borderColor": "#dc3545",
                "outlineColor": "#dc3545",
              }
            }
            title=""
          >
            Close
          </button>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="scroll-top-container"
  >
    <svg
      className="scrollTop"
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
          "display": "none",
          "height": 40,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 256C8 119 119 8 256 8s248 111 248 248-111 248-248 248S8 393 8 256zm143.6 28.9l72.4-75.5V392c0 13.3 10.7 24 24 24h16c13.3 0 24-10.7 24-24V209.4l72.4 75.5c9.3 9.7 24.8 9.9 34.3.4l10.9-11c9.4-9.4 9.4-24.6 0-33.9L273 107.7c-9.4-9.4-24.6-9.4-33.9 0L106.3 240.4c-9.4 9.4-9.4 24.6 0 33.9l10.9 11c9.6 9.5 25.1 9.3 34.4-.4z"
      />
    </svg>
  </div>,
]
`;
